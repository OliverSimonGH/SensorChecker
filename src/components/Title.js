import React, {Component} from 'react';

class Title extends Component {
    render() {
        return (
            <div className="page-header center">
                <h2>{this.props.title}</h2>
            </div>

        );
    }
}

export default Title;