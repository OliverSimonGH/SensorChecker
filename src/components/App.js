import React, {Component} from 'react';
import SensorSearch from "./SensorSearch";
import SensorData from "./SensorData";
import Title from "./Title";

import './styles/styles.css'
import {SENSOR_DATA} from '../events'

import openSocket from 'socket.io-client';

const socket = openSocket('http://localhost:8000');

class App extends Component {

    state = {
        sensor_data: [],
        filtered_data: [],
        searching: false
    }

    componentDidMount() {
        socket.on(SENSOR_DATA, (data) => {
            this.setState({
                sensor_data: [data, ...this.state.sensor_data]
            })
        });
    }

    searchChange = event => {
        if (event.target.value === "") {
            this.setState({
                searching: false
            })
        }
        else {
            this.setState({
                filtered_data: this.filterSensors(this.state.sensor_data, event.target.value, this.state.sensor_data.length),
                searching: true
            });
        }
    }

    filterSensors(data, searchText, maxResults) {
        return data.filter((data) => {
            return data.sensor.toUpperCase().includes(searchText.toUpperCase());
        }).slice(0, maxResults);
    }

    render() {
        return (
            <div className="container">
                <Title title="Sensor Application"/>
                <SensorSearch search_change={this.searchChange}/>
                {
                    this.state.searching &&
                    <SensorData sensor_data={this.state.filtered_data}/>
                }
                {
                    !this.state.searching &&
                    <SensorData sensor_data={this.state.sensor_data}/>
                }
            </div>
        );
    }
}

export default App;
