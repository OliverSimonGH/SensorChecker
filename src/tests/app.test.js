import React from 'react';
import {shallow, mount} from 'enzyme';
import App from '../components/App';
import Title from "../components/Title";
import renderer from 'react-test-renderer';

describe("App Component", () => {
    it('should count the correct amount of child components', function () {
        const wrapper = shallow(<App/>);
        const title = wrapper.find('Title');
        const search = wrapper.find('SensorSearch');
        const data = wrapper.find('SensorData');

        expect(title.length).toEqual(1);
        expect(search.length).toEqual(1);
        expect(data.length).toEqual(1);
    });

    it('matches the snapshot', () => {
        const tree = renderer.create(<App />).toJSON();
        expect(tree).toMatchSnapshot();
    })
})