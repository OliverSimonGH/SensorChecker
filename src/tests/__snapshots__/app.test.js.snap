// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component matches the snapshot 1`] = `
<div
  className="container"
>
  <div
    className="page-header center"
  >
    <h2>
      Sensor Application
    </h2>
  </div>
  <div
    className="mb15"
  >
    <input
      aria-label="Search Sensors"
      className="form-control"
      onChange={[Function]}
      placeholder="Search Sensors"
      type="text"
    />
  </div>
  <table
    className="table table-hover table-dark"
  >
    <thead>
      <tr>
        <th
          colSpan="4"
          scope="col"
        >
          Sensor Data
        </th>
      </tr>
      <tr>
        <th
          scope="col"
        >
          Index
        </th>
        <th
          scope="col"
        >
          Condition
        </th>
        <th
          scope="col"
        >
          Street
        </th>
        <th
          scope="col"
        >
          Value
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          colSpan="4"
        >
          No data available
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
